PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -1605,6 +1721,15 @@ void AliAnalysisTaskEmcalEmbeddingHelper::UserExec(Option_t*)
     AliError("Chain not initialized before running! Setting up now.");
     SetupEmbedding();
   }
+  else if(fEmbeddedRunblock.size()!=0)
+  {
+    // I do not think this is enough, I am not sure how to know if the train 
+    // was setup for run by run or run mixed analysis
+    AliAnalysisManager *mgr = AliAnalysisManager::GetAnalysisManager();
+    if ( mgr && fDataRunNumber > -1 && fDataRunNumber!=mgr->GetRunFromPath() )
+      AliError(Form("CAREFUL! Check what you are doing, you are embedding a data run %d that belongs to another runblock [%d,%d]!, setup the train to do the analysis per Run",
+                 InputEvent()->GetRunNumber(),fEmbeddedRunblockMin,fEmbeddedRunblockMax));
+  }
```

Comment:
If you remove unwanted filenames, I think this also isn't an issue anymore. This is hard to get right (I think even my solution isn't ideal, but at least it's reasonably well tested - I could imagine modify https://github.com/alisw/AliPhysics/blob/719ea3248a354a47a2082a510d98600d2d735bf4/PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx#L965-L968 to return early using the same mechanism as the internal event selection (here: https://github.com/alisw/AliPhysics/blob/719ea3248a354a47a2082a510d98600d2d735bf4/PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx#L1792-L1795). But first let's focus on the general design)